### PR TITLE
add react ssr document

### PR DESCRIPTION
**Pull request by iflamed**
According #467, We update the `react-renderer` to show how to support SSR with React.

### CODE REVIEW

**Comment by iflamed** [comment]
@yusukebe You may review this PR and then merge it.

File: packages/react-renderer/README.md [hunk 1]
```diff
@@ -6,11 +6,35 @@ React Renderer Middleware allows for the easy creation of a renderer based on Re
 
 ```txt
 npm i @hono/react-renderer react react-dom hono
-npm i -D @types/react @types/react-dom
+npm i -D @types/react @types/react-dom @vitejs/plugin-react
```
Comment (yusukebe) [inline packages/react-renderer/README.md:9]: I think `@vitejs/plugin-react` is not necessary. Can you remove it and try to run your app again?

Comment (iflamed) [reply packages/react-renderer/README.md:9]: It's need. Without this plugin, `vscode` will report error. And I checked the official demo from vite, they also have this plugin.

Comment (yusukebe) [reply packages/react-renderer/README.md:9]: Can you try this repo: https://github.com/yusukebe/hono-react-ssr/tree/vite? This may work.

Comment (iflamed) [reply packages/react-renderer/README.md:9]: I have tried this repo, no error found. Except loss the package dependency `@cloudflare/workers-types`

**Comment by yusukebe** [comment]
@iflamed 

I've added a minor change. Ready to merge now. Thanks!

**Pull request merged**